src/canal.erl

```diff
@@ -128,6 +130,10 @@ handle_call({write, Key, Body}, From, State) ->
 
 -spec handle_cast(_, state()) -> {noreply, state()}.
 
+handle_cast(reauth, State = #state{auth = #auth{token = Token}})
+    when Token =/= undefined ->
+    {noreply, State};
```

Comment:
I may be misreading the code but I'm not seeing where `#auth{token}` could be set back to `undefined`. This leads me to believe that this clause will sinkhole any auth attempt after the initial first successful auth.